Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -115,13 +343,38 @@ namespace Svc {
             return Fw::CmdResponse::EXECUTION_ERROR;
         }
 
+        // reset state file data
+        this->resetStateFileData();
+
+        // load state data from file
+        Fw::CmdResponse response = this->loadStateFile();
+
+        // reset free list for entries
+        this->resetBinaryTree();
+
+        // fill the binary tree with DP files
+        response = this->fillBinaryTree();
+        if (response != Fw::CmdResponse::OK) {
+            return response;
```

Comment:
Does this cause an inconsistent state? You may have failed with a partial binary tree, and an old or empty state file.   This return bypasses the state file prune and write in the next (nominal) step.